cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/Transactions.java

```diff
@@ -181,15 +182,34 @@ public Void call() throws Exception {
    * given {@link DynamicDatasetCache}.
    *
    * @param datasetCache The {@link DynamicDatasetCache} to use fo transaction creation as well as provided to the
-   *                     {@link TxRunnable} for access to dataset.
-   * @return a new instance of {@link Transactional}.
+   *                     {@link TxRunnable} for access to dataset
+   * @return a new instance of {@link Transactional}
    */
   public static Transactional createTransactional(final DynamicDatasetCache datasetCache) {
+    return createTransactional(datasetCache, null);
+  }
+
+  /**
+   * Creates a new instance of {@link Transactional} for {@link TxRunnable} execution using the
+   * given {@link DynamicDatasetCache}.
+   *
+   * @param datasetCache The {@link DynamicDatasetCache} to use fo transaction creation as well as provided to the
+   *                     {@link TxRunnable} for access to dataset
+   * @param defaultTimeout The default transaction timeout, used for starting the transaction for
+   *                       {@link Transactional#execute(TxRunnable)}
+   * @return a new instance of {@link Transactional}
+   */
+  public static Transactional createTransactional(final DynamicDatasetCache datasetCache,
+                                                  @Nullable final Integer defaultTimeout) {
     return new Transactional() {
       @Override
       public void execute(TxRunnable runnable) throws TransactionFailureException {
         TransactionContext txContext = datasetCache.newTransactionContext();
-        txContext.start();
+        if (defaultTimeout != null) {
```

Comment:
Seems like if we have the `defaultTimeout` not being null and defaulted to `data.tx.timeout` in AbstractContext, we can save this `if-else` logic in couple places.